website/static/js/folderPickerNodeConfig.js

```diff
@@ -0,0 +1,415 @@
+                xhr: xhr,
+                status: status,
+                error: error
+            });
+        }
+        return $osf.putJSON(self.urls().importAuth, {})
+            .done(onImportSuccess)
+            .fail(onImportError);
+    },
+    importAuth: function() {
+        /**
+         * Send PUT request to import access token from user profile.
+         */
+        var self = this;
+        bootbox.confirm({
+            title: 'Import ' + self.addonName + ' Access Token?',
+            message: self.messages.CONFIRM_AUTH(),
+            callback: function(confirmed) {
+                if (confirmed) {
+                    self._importAuthConfirm();
+                }
+            }
+        });
+    },
+    _deauthorizeConfirm: function(){
+        /**
+         * Send DELETE request to deauthorize this node.
+         */
+        var self = this;
+        var request = $.ajax({
+            url: self.urls().deauthorize,
+            type: 'DELETE'
+        });
+        request.done(function() {
+            // Update observables
+            self.nodeHasAuth(false);
+            self.cancelSelection();
+            self.currentDisplay(null);
+            self.changeMessage(self.messages.DEAUTHORIZE_SUCCESS(), 'text-warning', 3000);
+        });
+        request.fail(function(xhr, textStatus, error) {
+            self.changeMessage(self.messages.DEAUTHORIZE_FAIL(), 'text-danger');
+            Raven.captureMessage('Could not deauthorize ' + self.addonName + ' account from node', {
+                url: self.urls().deauthorize,
+                textStatus: textStatus,
+                error: error
+            });
+        });
+        return request;
+    },
+    deauthorize: function() {
+        /** Pop up a confirmation to deauthorize addon from this node.
+         *  Send DELETE request if confirmed.
+         */
+        var self = this;
+        bootbox.confirm({
+            title: 'Deauthorize ' + self.addonName + '?',
+            message: self.messages.CONFIRM_DEAUTH(),
+            callback: function(confirmed) {
+                if (confirmed) {
+                    self._deauthorizeConfirm();
+                }
+            }
+        });
+    },
+    cancelSelection: function() {
+        /**
+         * Must be used to update radio buttons and knockout view model simultaneously
+         */
+        this.selected(null);
+    },
+    togglePicker: function() {
+        /**
+         *  Toggles the visibility of the folder picker.
+         */
+        // Toggle visibility of folder picker
+        var shown = this.currentDisplay() === this.PICKER;
+        if (!shown) {
+            this.currentDisplay(this.PICKER);
+            this.activatePicker();
+        } else {
+            this.currentDisplay(null);
+            // Clear selection
+            this.cancelSelection();
+        }
+    },
+    treebeardOptions: function(){
```

Comment:
@sloria 

This allows subclasses to customize the Treebeard config.
The obvious alternative is to pass treebeardOptions to the constructor. 

Do you have a preference here, or have an alternative you'd rather see?